### DISCUSSION TITLE
problems pushing my local repository to by remote GitHub

### DISCUSSION THREAD

**Discussion by FlashJackFromGungagai** [community/community#45698] [New to GitHub] [answered]
Hello,
I am having problems pushing my local repository to by remote GitHub, with this:
1	I created a public GitHub for some software testing work that I am doing, and I have made it public.
2	This is at https://github.com/FlashJackFromGungagai/20230118ATMUnitTesting 
3	I have used IntelliJ IDEA to develop my testing scripts.
4	Using GitHub Desktop:
4.1	I have pointed it to the local repository
4.2	I have logged into my remote GitHub through GitHub Desktop.
4.3	I get the following message:

" You don't have write access to 20230118ATMUnitTesting. Want to create a fork?
commit to main"

I do not want to create a fork, and I do wish to commit to main:
1	I am still a novice with Git/GitHub and so I am not familiar with this message, is it telling me:
1.1	That there is a security on my remote GitHub setting that is preventing me from pushing from my local repository to my remote GitHub? If so, what setting do I need to change?
1.2	That there is a security setting on my local repository that is preventing me from pushing from my local repository to my remote GitHub? If so, what setting do I need to change

**Comment by surjendu104** [comment, marked as answer]
## You can do the following things
### 1. check you have read-write permission or not
    To do so,
    1. Go to your repository
    2. Click on the setting 
    3. Click the "Action" option on the left side bar
    4. Scroll down and check if you have the "Read and write permissions" checked under Workflow permissions
### 2. Initialize git repo in local folder and Push using git
    1. Make sure you have git installed in your system
    2. open terminal in your project folder and write these commands
    3. $ git init
    4. $ git add .
    5. $ git commit -m "<commit_message>"
    6. $ git remote add origin git@github.com:username/new_repo or $ git remote add origin https://github.com/github_username/repo_name.git
    7. $ git push -u origin master 
    If you have want to push in main branch then after  step-1 write this command-
        $ git checkout -b main
     then do rest of the steps
### 3. Alternatively you can upload the files manually using the "Add file" option in your git repository
![add file](https://user-images.githubusercontent.com/101238933/215320759-c06011b9-2ce6-46e1-aeba-173386fa9eb4.png)

You can also check these sites 
    https://forum.uipath.com/t/push-problems-not-write-access-to-the-repository/357133  
    https://kbroman.org/github_tutorial/pages/init.html
    https://docs.github.com/en/get-started/importing-your-projects-to-github/importing-source-code-to-github/adding-locally-hosted-code-to-github

**Reply by FlashJackFromGungagai** [reply]
Bingo that Worked...
To do so,
1. Go to your repository
2. Click on the setting 
3. Click the "Action" option on the left side bar
4. Scroll down and check if you have the "Read and write permissions" checked under Workflow permissions

Thank you. I have now been able to push using GitHub Desktop from my local to my remote repository.  Thank you for your detailed and rapid response.

**Reply by surjendu104** [reply]
Mark this one as answered if it helped you

**Reply by ghost** [reply]
I tried the same methos i can't able to push 
![Uploading Screenshot 2024-05-18 200023.png…]()

**Comment by airtower-luna** [comment]
> You don't have write access to 20230118ATMUnitTesting. Want to create a fork?

This sounds like you're not allowed to push to the repository on GitHub that you're trying to push to. However, you should always have write access to your own repository. So I guess you're either logged in as the wrong user, or pushing the wrong repository (maybe because you cloned from somewhere else, and that URL is still configured?). :thinking: 

I'm not sure how to check either of those in GitHub Desktop, on the command line you can check the remote repositories with `git remote -v`.

**Comment by jge162** [comment]
Does not look like you have correct permissions.

**Comment by DrakeYounger** [comment]
This is a security feature built into GitHub to prevent unauthorized changes to the codebase. Ensure you are logged into GitHub Desktop with the correct account with write access to the repository. Try creating a new branch in your local storage and pushing the changes to the new branch instead of the main one. Once the changes are on the new branch, you can create a pull request to merge them into the main branch. On a side note, the benefits of Agile Software Testing go beyond just having an [agile test environment](https://www.zaptest.com/what-is-agile-testing-process-life-cycle-methods-implementation). Agile methodologies can improve collaboration, increase efficiency, and produce higher-quality software overall.

**Comment by leproduction** [comment]
100% you can push file to github. https://www.youtube.com/watch?v=V8WbIERFOJU

**Comment by srivastava-jai** [comment]
I have enabled the read write permission but still getting error:-
remote: Permission to srivastava-jai/microservices-in-python.git denied to srivastava-jai.
fatal: unable to access 'https://github.com/srivastava-jai/microservices-in-python.git/': The requested URL returned error: 403

I have created PAT also did wverything I can but fail to push my clone repo's.

**Reply by surjendu104** [reply]
did you have logged in `git` in console.
To login to `git` by terminal follow [this thread ](https://stackoverflow.com/questions/57947268/how-do-i-log-in-using-the-git-terminal)

**Comment by srivastava-jai** [comment]
Hi surjendu,

It worked for me thanks for your help.

Regards
Jai Srivastava

**Comment by dcaayushd** [comment]


**Reply by Victor-Odunsi** [reply]
Hey 
Thanks for the steps

But they did not work for me.

I cannot push to Github as it returns  ! [remote rejected] main -> main (push declined due to repository rule violations)

Help

**Comment by ghost** [comment]
Hi Community/Community,
I am not getting manage access in github

**Comment by SohelMartin** [comment]
error: failed to push some refs to 'https://github.com/repo.git'
error: src refspec master does not match any

this errror are coming i cant push my code to my repo please help me

**Reply by Medhansh11** [reply]
generate a ssh key from cmd into your home directory. A file will be created copy the contents inside it and add it as a new SSH key in your github account.

**Comment by Joyce061998** [comment]
How to save this article? It will be a lot of help

**Comment by jayDevCodes** [comment]
First, verify your remote URL (git remote -v) and that you’re authenticated as the right user (SSH key or PAT).

If push is rejected, check repository permissions (are you a collaborator?) and branch protection rules (required status checks or review).

If using GitHub Actions to push, ensure the workflow has Workflow permissions → Read and write enabled in repo Settings so the runner can push.

Common fix: reconfigure credentials (generate a new PAT with repo scope) or ask repo admin to adjust permissions / branch rules.
